drivers/abp2/abp2_saul.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2024 CNRS, France
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     drivers_abp2
+ * @{
+ *
+ * @file
+ * @brief       ABP2 adaption to the SAUL framework.
+ *
+ * @author      David Picard <david.picard@clermont.in2p3.fr>
+ * @}
+ */
+
+#include <string.h>
+#include <stdio.h>
+
+#include "saul.h"
+#include "abp2.h"
+
+static int read_press(const void *dev, phydat_t *press)
+{
+    abp2_raw_t rawData;
+
+    if (abp2_measure_read((abp2_t *)dev, &rawData)) {
+        return -ECANCELED;
+    }
+
+    /* abp2_pressure() returns a value in thousandths of user units.
+     * here: mbar --> ubar */
```

Comment:
```suggestion
    /* abp2_pressure() returns a value in  user units. */
```

This might need a check if the actual value fits inside phydat.val -- this is conversion from 32bit integer to 16bit integer.

Reply:
int16_max is about  32000